From e4df658ea0751ce78598b8783664899b0ad54a52 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Sat, 26 Feb 2011 05:38:48 +0000
Subject: Add a desktop file for gnome-keyring-prompt

This is necessary for gnome-shell to pick up a meaningful
name and icon to display in the application menu while the
dialog is shown.
https://bugzilla.gnome.org/show_bug.cgi?id=643229
---
diff --git a/configure.in b/configure.in
index 3249666..8e03c7d 100644
--- a/configure.in
+++ b/configure.in
@@ -730,6 +730,7 @@ schema/Makefile
 testing/Makefile
 tool/Makefile
 ui/Makefile
+ui/gnome-keyring-prompt.desktop.in
 ui/tests/Makefile
 ])
 
diff --git a/po/POTFILES.in b/po/POTFILES.in
index a21a4f9..7485677 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -30,4 +30,5 @@ pkcs11/wrap-layer/gkm-wrap-login.c
 pkcs11/wrap-layer/gkm-wrap-prompt.c
 tool/gkr-tool.c
 ui/gku-prompt-tool.c
+ui/gnome-keyring-prompt.desktop.in.in
 [type: gettext/glade]ui/gku-prompt.ui
diff --git a/po/POTFILES.skip b/po/POTFILES.skip
index 4230965..4a6ef38 100644
--- a/po/POTFILES.skip
+++ b/po/POTFILES.skip
@@ -6,3 +6,4 @@ daemon/org.freedesktop.secrets.service.in
 daemon/org.gnome.keyring.service.in
 gcr/gcr-import-dialog.ui
 gcr/gcr-certificate-basics-widget.ui
+ui/gnome-keyring-prompt.desktop.in
diff --git a/ui/Makefile.am b/ui/Makefile.am
index 5073e46..ded3d50 100644
--- a/ui/Makefile.am
+++ b/ui/Makefile.am
@@ -84,3 +84,18 @@ if WITH_GTK3
 install-exec-hook:
 	@test -e "$(DESTDIR)$(libexecdir)/gnome-keyring-prompt" || (cd $(DESTDIR)$(libexecdir) && $(LN_S) gnome-keyring-prompt@GCR_VERSION_SUFFIX@ gnome-keyring-prompt)
 endif
+
+@INTLTOOL_DESKTOP_RULE@
+desktopdir= $(datadir)/applications
+desktop_in_in_files = gnome-keyring-prompt.desktop.in.in
+desktop_in_files = gnome-keyring-prompt.desktop.in
+desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
+
+EXTRA_DIST += \
+	$(desktop_in_in_files)
+
+CLEANFILES = \
+	$(desktop_DATA)
+
+DISTCLEANFILES = \
+	$(desktop_in_files)
diff --git a/ui/gnome-keyring-prompt.desktop.in.in b/ui/gnome-keyring-prompt.desktop.in.in
new file mode 100644
index 0000000..917b5a0
--- a/dev/null
+++ b/ui/gnome-keyring-prompt.desktop.in.in
@@ -0,0 +1,8 @@
+[Desktop Entry]
+_Name=Keyring Access
+_Comment=Unlock access to passwords and other secrets
+Icon=security-medium
+Exec=@LIBEXECDIR@/gnome-keyring-prompt
+Terminal=false
+Type=Application
+NoDisplay=true
--
cgit v0.8.3.4
